test/handlers/generic/executeDeposit.js

```diff
@@ -0,0 +1,150 @@
+            centrifugeAssetFuncSig.substr(2) + hashOfCentrifugeAsset.substr(2)
+        depositProposalDataHash = Ethers.utils.keccak256(GenericHandlerInstance.address + depositProposalData.substr(2));
+    });
+
+    it('deposit can be executed successfully', async () => {
+        TruffleAssert.passes(await BridgeInstance.deposit(
+            chainID,
+            GenericHandlerInstance.address,
+            depositData,
+            { from: depositerAddress }
+        ));
+
+        // relayer1 creates the deposit proposal
+        TruffleAssert.passes(await BridgeInstance.voteDepositProposal(
+            chainID,
+            expectedDepositNonce,
+            depositProposalDataHash,
+            { from: relayer1Address }
+        ));
+
+        // relayer2 votes in favor of the deposit proposal
+        // because the relayerThreshold is 2, the deposit proposal will go
+        // into a finalized state
+        TruffleAssert.passes(await BridgeInstance.voteDepositProposal(
+            chainID,
+            expectedDepositNonce,
+            depositProposalDataHash,
+            { from: relayer2Address }
+        ));
+
+        // relayer1 will execute the deposit proposal
+        TruffleAssert.passes(await BridgeInstance.executeDepositProposal(
+            chainID,
+            expectedDepositNonce,
+            GenericHandlerInstance.address,
+            depositProposalData
+        ));
+        
+        // Verifying asset was marked as stored in CentrifugeAssetInstance
+        assert.isTrue(await CentrifugeAssetInstance._assetsStored.call(hashOfCentrifugeAsset));
+    });
+
+    it('AssetStored event should be emitted', async () => {
+        TruffleAssert.passes(await BridgeInstance.deposit(
+            chainID,
+            GenericHandlerInstance.address,
+            depositData,
+            { from: depositerAddress }
+        ));
+
+        // relayer1 creates the deposit proposal
+        TruffleAssert.passes(await BridgeInstance.voteDepositProposal(
+            chainID,
+            expectedDepositNonce,
+            depositProposalDataHash,
+            { from: relayer1Address }
+        ));
+
+        // relayer2 votes in favor of the deposit proposal
+        // because the relayerThreshold is 2, the deposit proposal will go
+        // into a finalized state
+        TruffleAssert.passes(await BridgeInstance.voteDepositProposal(
+            chainID,
+            expectedDepositNonce,
+            depositProposalDataHash,
+            { from: relayer2Address }
+        ));
+
+        // relayer1 will execute the deposit proposal
+        const executeDepositTx = await BridgeInstance.executeDepositProposal(
+            chainID,
+            expectedDepositNonce,
+            GenericHandlerInstance.address,
+            depositProposalData
+        );
+        
+        TruffleAssert.eventNotEmitted(executeDepositTx, 'AssetStored', event => {
```

Comment:
The event should be emitted here, but TruffleAssert does not recognize the event as being part of the `executeDepositTx`

Reply:
Seems to be a limitation of truffle-assertions: https://github.com/rkalis/truffle-assertions/issues/6

This works:
```js
let internalTx = await TruffleAssert.createTransactionResult(CentrifugeAssetInstance, executeDepositTx.tx);
TruffleAssert.eventEmitted(internalTx, 'AssetStored', event => {
    return event.asset === hashOfCentrifugeAsset;
})
```

Reply:
do you know why?

Reply:
...the issue I linked?